scripts/saccade.py

```diff
@@ -57,21 +58,24 @@ def __init__(self):
         self.M               = np.zeros(self.N)            # movement neurons
         self.V               = np.zeros(self.N)            # visual neurons
     
-        rospy.Subscriber("/saliency_map", Image, self.saliency_callback, queue_size=1, buff_size=2**24)
+        self.saliency_sub = rospy.Subscriber("/saliency_map", Float32MultiArray, self.saliency_map_callback, queue_size=1, buff_size=2**24)
 
         self.target_pub = rospy.Publisher("/saccade_target", Point, queue_size=1)
         self.potential_target_pub = rospy.Publisher("/saccade_potential_target", Point, queue_size=1)
 
         self.cv_bridge = CvBridge()
 
+        self.reset_saccade_serv = rospy.Service('/reset_saccade', ResetSaccade, self.handle_reset_saccade)
+
     # numerical integration (simple Euler)
-    def saliency_callback(self, saliency_map):
+    def saliency_map_callback(self, saliency_map):
 
         # handle input
-        try:
-            sal = self.cv_bridge.imgmsg_to_cv2(saliency_map, "mono8")
-        except CvBridgeError as e:
-            print e
+        lo = saliency_map.layout
```

Comment:
What about the single liner:
``sal = saliency_map.data[lo.data_offset:].reshape(lo.dim[0].size, lo.dim[1].size)``